run/testing/markdown_preview_renderer.e2e_test.go

```diff
@@ -52,31 +58,35 @@ func TestRendererService(t *testing.T) {
 	}
 
 	for _, test := range tests {
-		req, err := service.NewRequest("POST", "/")
-		if err != nil {
-			t.Fatalf("service.NewRequest: %q", err)
-		}
-		req.Body = ioutil.NopCloser(strings.NewReader(test.input))
+		testutil.Retry(t, 5, 30*time.Second, func(r *testutil.R) {
+			req, err := service.NewRequest("POST", "/")
+			if err != nil {
+				r.Errorf("service.NewRequest: %q", err)
+				return
+			}
+			req.Body = ioutil.NopCloser(strings.NewReader(test.input))
 
-		client := http.Client{Timeout: 10 * time.Second}
-		resp, err := client.Do(req)
-		if err != nil {
-			t.Fatalf("client.Do: %v", err)
-		}
-		defer resp.Body.Close()
-		t.Logf("client.Do: %s %s\n", req.Method, req.URL)
+			client := http.Client{Timeout: 10 * time.Second}
+			resp, err := client.Do(req)
+			if err != nil {
+				r.Errorf("client.Do: %v", err)
+				return
+			}
+			defer resp.Body.Close()
+			r.Logf("client.Do: %s %s\n", req.Method, req.URL)
 
-		if got := resp.StatusCode; got != http.StatusOK {
-			t.Errorf("response status: got %d, want %d", got, http.StatusOK)
-		}
+			if got := resp.StatusCode; got != http.StatusOK {
+				r.Errorf("response status: got %d, want %d", got, http.StatusOK)
+			}
 
-		out, err := ioutil.ReadAll(resp.Body)
-		if err != nil {
-			t.Fatalf("ioutil.ReadAll: %v", err)
-		}
+			out, err := ioutil.ReadAll(resp.Body)
```

Comment:
nit: I see that current Go style is to use `io` rather than `ioutil`.

See:
https://github.com/golang/go/issues/42026